dot/core/service.go

```diff
@@ -372,17 +380,33 @@ func (s *Service) maintainTransactionPool(block *types.Block) {
 		s.transactionState.RemoveExtrinsic(ext)
 	}
 
+	bestBlockHash := s.blockState.BestBlockHash()
+	stateRoot, err := s.storageState.GetStateRootFromBlock(&bestBlockHash)
+	if err != nil {
+		logger.Errorf("could not get state root from block %s: %w", bestBlockHash, err)
+	}
+
+	ts, err := s.storageState.TrieState(stateRoot)
+	if err != nil {
+		logger.Errorf(err.Error())
```

Comment:
Shouldn't we return an error instead here?

Reply:
Also Im thinking so here. Will still wait for more feedback to double check

Reply:
Same here, how can we continue if we don't have the trie state?